pkg/types/nutanix/validation/platform.go

```diff
@@ -10,39 +12,100 @@ import (
 	}
-	if len(p.Password) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("password"), "must specify the password"))
+
+	if len(p.PrismCentral.Password) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismCentral").Child("password"),
+			"must specify the Prism Central password"))
 	}
-	if len(p.Port) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("port"), "must specify the port"))
+
+	if len(p.PrismElements) != 1 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements"), "must specify the Prism Element (cluster)"))
 	}
-	if len(p.SubnetUUID) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("subnet"), "must specify the subnet"))
+
+	if len(p.PrismElements[0].Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements[0]").Child("name"),
+			"must specify the Prism Element (cluster) name"))
 	}
-	if len(p.PrismCentral) != 0 {
-		if err := validate.Host(p.PrismCentral); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("prismCentral"), p.PrismCentral, "must be the domain name or IP address of the Prism Central"))
+
+	if len(p.PrismElements[0].Endpoint.Address) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements[0]").Child("endpoint").Child("address"),
+			"must specify the Prism element (cluster) endpoint address"))
+	} else {
+		if err = validate.Host(p.PrismElements[0].Endpoint.Address); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("prismElements[0]").Child("endpoint").Child("address"),
+				p.PrismElements[0].Endpoint.Address, "must be the domain name or IP address of the Prism Element (cluster)"))
 		}
 	}
 
+	if p.PrismElements[0].Endpoint.Port < 1 || p.PrismElements[0].Endpoint.Port > 65535 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements[0]").Child("endpoint").Child("port"),
+			"The Prism Element (cluster) endpoint port is invalid, must be in the range of 1 to 65535"))
+	}
+
+	if len(p.PrismElements[0].Username) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements[0]").Child("username"),
+			"must specify the Prism Element (cluster) username"))
+	}
+
+	if len(p.PrismElements[0].Password) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements[0]").Child("password"),
+			"must specify the Prism Element (cluster) password"))
+	}
+
+	//if len(p.DefaultStorageContainer) == 0 {
+	//	allErrs = append(allErrs, field.Required(fldPath.Child("defaultStorageContainer"), "must specify the default storage container"))
+	//}
+
+	if len(p.SubnetUUID) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("subnet"), "must specify the subnet"))
+	}
+
 	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
-	if p.APIVIP != "" || p.IngressVIP != "" {
+	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
```

Comment:
Don't make this change. This was rejected in the initial PR.

Reply:
Reversed this unexpected change.